demo/clusterctl-examples/sample-kubeadm-ctlplane-crio-ubuntu.rc

```diff
@@ -0,0 +1,260 @@
+    - name: metal3
+      shell: /bin/bash
+      sshAuthorizedKeys:
+      - ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAICXtmbErb2jcl7iVZToyq+IRSgILaPC0V99C1pSsx83S
+      sudo: ALL=(ALL) NOPASSWD:ALL
+"
+
+#
+# Commands to be executed after *kubeadm init* is successfully executed.
+#
+#
+# NOTE: these are YAML snippets to be injected into kubeadmConfigSpec so
+# make sure we keep the 4 spaces indentation for each line.
+#
+export KUBEADM_CONTROL_PLANE_POST_KUBEADM_COMMANDS="
+    - mkdir -p /home/metal3/.kube
+    - chown metal3:metal3 /home/metal3/.kube
+    - cp /etc/kubernetes/admin.conf /home/metal3/.kube/config
+    - systemctl enable --now keepalived
+    - chown metal3:metal3 /home/metal3/.kube/config
+"
+
+#
+# Commands to be executed prior to running *kubeadm init*. For a base image
+# which has nothing on it. These commands are responsible for installing
+# the necessary binaries to bootstrap the control plane.
+#
+# NOTE: these are YAML snippets to be injected into kubeadmConfigSpec so
+# make sure we keep the 4 spaces indentation for each line.
+#
+export KUBEADM_CONTROL_PLANE_PRE_KUBEADM_COMMANDS="
+    - netplan apply
+    - apt update -y
+    - apt install apt-transport-https ca-certificates curl gnupg-agent software-properties-common
+      -y
+    - curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add -
+    - add-apt-repository 'deb [arch=amd64] https://download.docker.com/linux/ubuntu
+      jammy stable' -y
+    - curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add
+      -
+    - echo 'deb https://apt.kubernetes.io/ kubernetes-xenial main' > /etc/apt/sources.list.d/kubernetes.list
+    - echo 'deb https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_22.04/ /'|sudo tee /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list
+    - echo 'deb http://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable:/cri-o:/1.23/xUbuntu_22.04/ /'|sudo tee /etc/apt/sources.list.d/devel:kubic:libcontainers:stable:cri-o:1.23.list
+    - curl -L https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable:cri-o:1.23/xUbuntu_22.04/Release.key | sudo apt-key add -
+    - curl -L https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_22.04/Release.key | sudo apt-key add -
+    - apt update -y
+    - apt install docker-ce docker-ce-cli containerd.io kubelet kubeadm kubectl cri-o cri-o-runc cri-tools
+      keepalived -y
+    - cp /etc/containers/registries.conf /etc/containers/registries.conf.orig
+    - cp /etc/containers/registries.conf.cp /etc/containers/registries.conf
+    - systemctl enable --now docker kubelet crio.service
+    - usermod -aG docker metal3
+    - if (curl -sk --max-time 10 https://{{ example_cluster_virtual_ip }}:6443/healthz); then echo
+      'keepalived already running';else systemctl start keepalived; fi
+    - systemctl enable --now /lib/systemd/system/monitor.keepalived.service
+"
```

Comment:
This is a nit but let's be consistent about line breaks. I'd suggest make them all one liners.

Reply:
Sounds good.